client/header/activity-panel/panels/inbox/action.js

```diff
@@ -64,13 +64,20 @@ class InboxNoteAction extends Component {
 		const { action, dismiss, label } = this.props;
 		const isPrimary = dismiss || action.primary;
 
+		let actionHref;
+		if ( action ) {
+			if ( action.url ) {
+				actionHref = action.url;
+			}
+		}
+
 		return (
 			<Button
 				isPrimary={ isPrimary }
 				isSecondary={ ! isPrimary }
 				isBusy={ this.state.inAction }
 				disabled={ this.state.inAction }
-				href={ action ? action.url : undefined }
+				href={ actionHref }
```

Comment:
Hey @octaedro  I wonder, could we make this code a little more succinct by removing the nested if on line 67 and use a ternary here instead. I think that although the ternary is a little long its a bit easier to understand quickly in this case. 😃 

```suggestion
				href={action && action.url && action.url.length ? action.url : undefined}
```

Reply:
Nice, this check for length of string is a good improvement and more thorough .

Reply:
Hi @psealock and @samueljseay, thank you for your review. 
Nice catch Sam! The change you proposed sounds very good 😀
I did it in the commit 98aba7c